at-quannguyen/W4/ex2.rb

```diff
@@ -0,0 +1,30 @@
+def add_more(num)
+  ->(x) { x + num }
+end
+
+def multi_more(num)
+  ->(x) { x * num }
+end
+
+def compose(proc1, proc2)
+  ->(x) { proc2.(proc1.(x)) }
+end
+
+add4 = add_more(4)
+add10 = add_more(10)
+
+multi10 = multi_more(10)
+multi5 = multi_more(5)
+
+add4_and_multi10 = compose(add4, multi10)
+multi10_and_add4 = compose(multi10, add4)
+
+p add4.(5)
+p add10.(12)
+
+p multi10.(10)
+p multi5.(10)
+
+p add4_and_multi10.(10)
+p multi10_and_add4.(10)
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/atruby-team/internship-18-summer/blob/cd24ce24c04c57d4b75cc24b8a37d518f57e5e0e/at-quannguyen/W4/ex2.rb#L29"></span>Style/LambdaCall: Prefer the use of <code>lambda.call(...)</code> over <code>lambda.(...)</code>. (https://github.com/rubocop-hq/ruby-style-guide#proc-call)</td>
    </tr>
  </tbody>
</table>